File: 0 esercitazione Lettura e Scrittura File in Java e C/my/svolta/C.java
import java.io.*;

public class C {

  public static void main(String[] args) {
    FileReader fr = null;
    String filename = null;
    int n;
    //controllo argomento
    if (args.length != 1) usage(); else filename = args[0];

    try {
      fr = new FileReader(filename);
      while ((n = fr.read()) >= 0) {
        //-1 means EOF
        System.out.print((char) n);
      }
      fr.close();
    } catch (FileNotFoundException fne) {
      fne.printStackTrace();
      System.exit(1);
    } catch (IOException ioe) {
      ioe.printStackTrace();
      System.exit(2);
    }
  }

  private static void usage() {
    System.out.println("Usage: C <filename>");
    System.exit(0);
  }
}

File: 0 esercitazione Lettura e Scrittura File in Java e C/my/svolta/C.c
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#define MAX_STRING_LENGTH 256
void usage()
{
    printf("Usage: C <filename>\n");
    exit(0);
}
int main(int argc, char **argv)
{
    int fd, nc;
    char *filename;
    char ch;
    // controllo paramentri
    if (argc != 2)
        usage();
    else
        filename = argv[1];
    fd = open(filename, O_RDONLY);
    if (fd < 0)
    {
        perror("Apertura file fallita");
        return EXIT_FAILURE;
    }
    while ((nc = read(fd, &ch, 1)) > 0)
    {
        if (nc == 1)
            putchar(ch);
        else
        {
            printf("PID %d: Errore nel read file %s", getpid(), filename);
            perror("Errore!");
            close(fd);
        }
    }
    close(fd);
    return EXIT_SUCCESS;
}
